designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/image/ZoomableImage.kt

```diff
@@ -63,7 +65,16 @@ internal fun ZoomableImage(
                 detectTapGestures(
                     onDoubleTap = { tapOffset ->
                         isDoubleTapped = true
-                        scale = if (scale > 1f) 1f else 2f
+                        scale = if (scale > 1f) {
+                            1f
+                        } else {
+                            val srcSize = Size(painter.intrinsicSize.width, painter.intrinsicSize.height)
+                            val scaleFactor = ContentScale.FillBounds.computeScaleFactor(srcSize, size.toSize())
+                            maxOf(
+                                maxOf(scaleFactor.scaleX, scaleFactor.scaleY) / minOf(scaleFactor.scaleX, scaleFactor.scaleY),
+                                2f,
+                            )
+                        }
```

Comment:
Can we extract this out into a separate variable and define what it does? It's hard to read.

Also, we can use `coerceAtMost` to clip the value to 1 instead of if else check.

Reply:
@msasikanth Fixed.